### PR TITLE
Use TestContainer for unit tests

### PR DESCRIPTION
**Pull request by phiz71**
**Issue**

https://gravitee.atlassian.net/browse/APIM-488

**Description**

 - Use TestContainer for unit tests
 - Move all Freemarker templates from reporter-elastic, so they can be used in repository-elastic 

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `4.1.0-testcontainer-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/elasticsearch/gravitee-common-elasticsearch/4.1.0-testcontainer-SNAPSHOT/gravitee-common-elasticsearch-4.1.0-testcontainer-SNAPSHOT.zip)

### CODE REVIEW

**Comment by gcusnieux** [comment]
Cool stuff ! But could you run tests on each version in ci ?

File: src/main/java/io/gravitee/elasticsearch/client/http/HttpClient.java [hunk 1]
```diff
@@ -267,16 +267,20 @@ public Single<Health> getClusterHealth() {
...
+        if (forceRefresh) {
+            bulkURL += "?refresh=true";
+        }
         return nextClient()
             .getClient()
-            .post(URL_BULK)
+            .post(bulkURL)
             .putHeader(HttpHeaders.CONTENT_TYPE, "application/x-ndjson")
             .rxSendBuffer(payload)
+            .doOnError(throwable -> logger.error("Unable to get bulk data to Elasticsearch: {}", throwable.getMessage()))
```
Comment (ytvnr) [inline src/main/java/io/gravitee/elasticsearch/client/http/HttpClient.java:283]:
```suggestion
            .doOnError(throwable -> logger.error("Unable to get bulk data from Elasticsearch: {}", throwable.getMessage()))
```

File: src/test/java/io/gravitee/elasticsearch/client/HttpClientTest.java [hunk 2]
```diff
@@ -45,6 +46,10 @@
 @ContextConfiguration(classes = { HttpClientTest.TestConfig.class })
 public class HttpClientTest {
 
+    public static final String ELASTICSEARCH_VERSION = "7.17.7";
```
Comment (ytvnr) [inline src/test/java/io/gravitee/elasticsearch/client/HttpClientTest.java:49]:
```suggestion
    public static final String ELASTICSEARCH_DEFAULT_VERSION = "7.17.7";
```

File: src/test/java/io/gravitee/elasticsearch/client/HttpClientTest.java [hunk 3]
```diff
@@ -113,65 +121,42 @@ public void shouldNotGetAlias() throws InterruptedException {
...
-        public HttpClientConfiguration configuration(ElasticsearchNode elasticsearchNode) {
+        public HttpClientConfiguration configuration(ElasticsearchContainer elasticSearchContainer) {
             HttpClientConfiguration elasticConfiguration = new HttpClientConfiguration();
             elasticConfiguration.setEndpoints(
-                Collections.singletonList(new Endpoint("http://localhost:" + elasticsearchNode.getHttpPort()))
+                Collections.singletonList(new Endpoint("http://" + elasticSearchContainer.getHttpHostAddress()))
             );
+            elasticConfiguration.setUsername("elastic");
+            elasticConfiguration.setPassword(ElasticsearchContainer.ELASTICSEARCH_DEFAULT_PASSWORD);
             //            elasticConfiguration.setIngestPlugins(Arrays.asList("geoip"));
```
Comment (ytvnr) [inline src/test/java/io/gravitee/elasticsearch/client/HttpClientTest.java:148]: can be deleted ?

**Review by ytvnr — APPROVED** [review]
looks good to me, minor remarks

**Pull request merged**